spring-integration-redis/src/test/java/org/springframework/integration/redis/util/RedisLockRegistryTests.java

```diff
@@ -97,13 +100,15 @@ public void testLock() throws Exception {
 			Lock lock = registry.obtain("foo");
 			lock.lock();
 			try {
-				assertNotNull(TestUtils.getPropertyValue(registry, "hardThreadLocks", ThreadLocal.class).get());
+				assertEquals(1, TestUtils.getPropertyValue(registry, "locks", Map.class).size());
 			}
 			finally {
 				lock.unlock();
 			}
 		}
-		assertNull(TestUtils.getPropertyValue(registry, "hardThreadLocks", ThreadLocal.class).get());
+		Thread.sleep(10);
+		registry.expireUnusedOlderThan(0);
```

Comment:
`registry.expireUnusedOlderThan(-1000);` instead of `Thread.sleep()` 😉 
I don't like any blocking in tests for nothing.